cmd/opampsupervisor/supervisor/supervisor_test.go

```diff
@@ -55,6 +59,6 @@ func Test_composeEffectiveConfig(t *testing.T) {
 	require.NoError(t, err)
 	expectedConfig = bytes.ReplaceAll(expectedConfig, []byte("\r\n"), []byte("\n"))
 
-	require.True(t, configChanged)
+	require.False(t, configChanged)
```

Comment:
I know the `composeEffectiveConfig` function was updated, but it's not immediately obvious to me why this changed. Do you know why this is different now?

Reply:
The offered config update in line 51 is empty and doesn't change anything. I forgot why the current logic is returning true but let me check and get back. I think the updated behaviour is correct.

Reply:
Thanks for digging into it. I think we should probably update the test since that flow involves more logic.